sklearn/utils/estimator_checks.py

```diff
@@ -2470,3 +2482,36 @@ def check_fit_idempotent(name, estimator_orig):
         if hasattr(estimator, method):
             new_result = getattr(estimator, method)(X_test)
             assert_allclose_dense_sparse(result[method], new_result)
+
+
+def check_outlier_resamplers(name, estimator_orig):
+    X, y = make_blobs(random_state=0)
+    outliers = estimator_orig.fit_predict(X, y) == -1
+    n_outliers = np.sum(outliers)
+
+    X_new, y_new = estimator_orig.fit_resample(X, y)
+
+    assert X_new.shape[0] == X.shape[0] - n_outliers
+    assert y_new.shape[0] == y.shape[0] - n_outliers
+
+
+def check_resampler_structure(name, estimator_orig):
+    X, y = make_blobs(n_samples=10)
+    X_new, y_new = estimator_orig.fit_resample(X, y)
+
+
+def check_resample_repeated(name, estimator_orig):
+    X, y = make_blobs(n_samples=10)
```

Comment:
Do we have enough samples to be sure that our resamplers are doing something here?